### PR TITLE
Fix/persisted native message system

### LINKED ISSUE #10979: Message system state is not presistent
**Describe the bug**
If is a closable message banner closed, it should never appear again.
**How to reproduce**

1. Close a banner message
1. Kill the app
1. Open the app again
1. See the banner message again

**Expected behavior**
Whole message system redux state is persisted.
**Screenshots**

https://github.com/trezor/trezor-suite/assets/26143964/cdd19612-e2de-4a2c-918b-55115a29ec12

### PR DESCRIPTION
**Pull request by PeKne**
## Description
- 9c69ee0b743b42cb4fb3986f2416b16570505f2d: message system made persistent 
- 730251c38a0faaee8327ed0d1addabe09b751d5d: messages validate on every device connection/selection

## Related Issue

Resolve #10979

### CODE REVIEW

File: suite-native/message-system/src/messageSystemMiddleware.ts [hunk 1]
```diff
@@ -5,18 +5,28 @@ import {
...
     getValidMessages,
+    selectMessageSystemConfig,
 } from '@suite-common/message-system';
-import { selectAccountsSymbols } from '@suite-common/wallet-core';
+import { deviceActions, selectAccountsSymbols, selectDevice } from '@suite-common/wallet-core';
 
-const isAnyOfMessageSystemAffectingActions = isAnyOf(messageSystemActions.fetchSuccessUpdate);
+const isAnyOfMessageSystemAffectingActions = isAnyOf(
+    messageSystemActions.fetchSuccessUpdate,
+    deviceActions.selectDevice,
```
Comment (vytick) [inline suite-native/message-system/src/messageSystemMiddleware.ts:14]: It seems, that there are also other actions that maybe should trigger the check - deviceChange, disconnect, forget, auth. Wdyt?

Comment (PeKne) [reply suite-native/message-system/src/messageSystemMiddleware.ts:14]: cc @matejkriz

Comment (matejkriz) [reply suite-native/message-system/src/messageSystemMiddleware.ts:14]: - deviceChanged and auth are not relevant (no relevant change for available message system conditions can be updated)
- disconnect and forget would also trigger selectDevice if selected, but not relevant if not selected

Comment (matejkriz) [reply suite-native/message-system/src/messageSystemMiddleware.ts:14]: but definitelly I'm glad you asked 👍 ❤️

**Review by matejkriz — COMMENTED** [review]
FIXED in https://github.com/trezor/trezor-suite/pull/11039/commits/1b5794b0cfcbd4f4a6d6318c62179c245a7a3ef6

---

Have you found a way to test it? I've tried to test it locally with custom local config (simulating remote fetch failed) and I'm now facing this error:

```
 ERROR  [Error: Unable to resolve module ./suite-common/message-system/files/config.v1 from /Users/matejkriz/Projects/trezor-suite/suite-native/app/.:

None of these files exist:
  * suite-common/message-system/files/config.v1(.android.ts|.native.ts|.ts|.android.tsx|.native.tsx|.tsx|.android.mjs|.native.mjs|.mjs|.android.js|.native.js|.js|.android.jsx|.native.jsx|.jsx|.android.json|.native.json|.json|.android.cjs|.native.cjs|.cjs|.android.scss|.native.scss|.scss|.android.sass|.native.sass|.sass|.android.css|.native.css|.css)
  ```
  
  But I believe it should work... https://github.com/trezor/trezor-suite/pull/7936#discussion_r1217744168

**Comment by matejkriz** [comment]
@tomasklim please check https://github.com/trezor/trezor-suite/pull/11039/commits/1b5794b0cfcbd4f4a6d6318c62179c245a7a3ef6 I know it's not great, but hopefully not terrible?

**Review by matejkriz — APPROVED** [review]
device condition works nice for me 
image

**Review by matejkriz — APPROVED** [review]
device condition works nice for me 
image

**Pull request merged**